### PR TITLE
Add LiteX I2C driver

### PR DESCRIPTION
**Pull request by mateusz-holenko**
This adds LiteX I2C bit-banged-based driver and enables it in litex_vexriscv board.

### CODE REVIEW

File: CODEOWNERS
Comment (galak) [inline CODEOWNERS]: Maybe just make this `/drivers/i2c/*litex*`

File: dts/riscv/riscv32-litex-vexriscv.dtsi
Comment (galak) [inline dts/riscv/riscv32-litex-vexriscv.dtsi]: We've historically not use'd 'L' in the hex values.  So having this just be `<0xe0003000 0x3 0xe0003004 0x1>

**Review by galak — COMMENTED** [review]
DTS stuff looks generally good, minor nits.

File: CODEOWNERS
Comment (mateusz-holenko) [reply CODEOWNERS]: Changed.

File: dts/riscv/riscv32-litex-vexriscv.dtsi
Comment (mateusz-holenko) [reply dts/riscv/riscv32-litex-vexriscv.dtsi]: Changed.

**Comment by galak** [comment]
DT stuff looks good now

**Comment by mateusz-holenko** [comment]
I fixed base address in the DTS.

**Comment by mateusz-holenko** [comment]
@galak The code is rebased on top of the 2.0.0 release. Is there anything else I should do before merging this PR?

**Comment by mateusz-holenko** [comment]
Rebased to resolve conflicts after merging #18649.

**Comment by mateusz-holenko** [comment]
@galak Can this PR be merged?

**Comment by mateusz-holenko** [comment]
I rebased the branch onto the newest master to resolve conflicts. @galak Is there anything else blocking this PR from being merged?

File: drivers/i2c/i2c_litex.c
Comment (kgugala) [inline drivers/i2c/i2c_litex.c]: there should be an empty line here

**Review by kgugala — CHANGES_REQUESTED** [review]
Minor code style issue, otherwise LGTM

File: drivers/i2c/i2c_litex.c
Comment (pabigot) [reply drivers/i2c/i2c_litex.c]: Also use `BIT(bit)` instead of shifting a signed integer, which produces undefined behavior `bit` is 31.

Please do that throughout the driver.

File: drivers/i2c/i2c_litex.c
Comment (anangl) [inline drivers/i2c/i2c_litex.c]: You could take `struct i2c_bitbang bitbang` out of `struct i2c_litex_cfg` and point it as the `driver_data` of the device, making use of this parameter. Then, `i2c_litex_cfg_##n` could be made `const`.

File: drivers/i2c/i2c_litex.c
Comment (mateusz-holenko) [reply drivers/i2c/i2c_litex.c]: Fixed

File: drivers/i2c/i2c_litex.c
Comment (mateusz-holenko) [reply drivers/i2c/i2c_litex.c]: Reworked as suggested

File: drivers/i2c/i2c_litex.c
Comment (anangl) [inline drivers/i2c/i2c_litex.c]: `const struct i2c_litex_cfg *`

File: drivers/i2c/i2c_litex.c
Comment (anangl) [inline drivers/i2c/i2c_litex.c]: This register is only read, and it seems to be not supposed to be written (at least through this pointer), so the pointer should be `volatile const u32_t *`.

File: drivers/i2c/i2c_litex.c
Comment (anangl) [inline drivers/i2c/i2c_litex.c]: `const struct i2c_litex_cfg *`, more occurrences below.

File: drivers/i2c/i2c_litex.c
Comment (anangl) [inline drivers/i2c/i2c_litex.c]: This should be `static`. Similarly all the other functions defined in this file. There's no need for their names to be seen outside.

**Review by anangl — APPROVED** [review]

**Review by kgugala — APPROVED** [review]

File: drivers/i2c/i2c_litex.c
Comment (mateusz-holenko) [reply drivers/i2c/i2c_litex.c]: Changed as suggested for all functions in the file

File: drivers/i2c/i2c_litex.c
Comment (mateusz-holenko) [reply drivers/i2c/i2c_litex.c]: Changed as suggested here and in other places

File: drivers/i2c/i2c_litex.c
Comment (mateusz-holenko) [reply drivers/i2c/i2c_litex.c]: Changed as suggested

File: drivers/i2c/i2c_litex.c
Comment (mateusz-holenko) [reply drivers/i2c/i2c_litex.c]: Changed as suggested

**Pull request merged**
